e2e/test/scenarios/dashboard/dashboard.cy.spec.js

```diff
@@ -683,6 +683,59 @@ describe("scenarios > dashboard", () => {
       cy.findByText("There was a problem displaying this chart.");
     });
   });
+
+  it("should not have markdown content overflow in description (metabase#31326)", () => {
+    cy.intercept("GET", "/api/dashboard/1").as("getDashboard");
+    cy.intercept("PUT", "/api/dashboard/1").as("updateDashboard");
+    visitDashboard(1);
+    cy.wait("@getDashboard");
+
+    cy.get("main header").icon("info").click();
+
+    const testMarkdownContent =
+      "{selectall}# Heading 1{enter}{enter}**bold** https://www.metabase.com/community_posts/how-to-measure-the-success-of-new-product-features-and-why-it-is-important{enter}{enter}![alt](/app/assets/img/welcome-modal-2.png){enter}{enter}This is my description. ";
+
+    rightSidebar().within(() => {
+      cy.findByPlaceholderText("Add description")
+        .click()
+        .type(testMarkdownContent)
+        .blur();
+    });
+
+    cy.wait("@updateDashboard");
+
+    rightSidebar().within(() => {
+      // check that markdown content is not bigger than its container
+      cy.findByTestId("editable-text").then($markdown => {
+        const el = $markdown[0];
+
+        // vertical
+        expect($markdown[0].clientHeight).to.be.gte(
+          $markdown[0].firstElementChild.clientHeight,
+        );
+
+        // horizontal
+        $markdown.find("*").each((index, childEl) => {
+          expect(el.clientWidth).to.be.gte(
+            childEl.getBoundingClientRect().width,
+          );
+        });
```

Comment:
A child can overflow its parent even if the parent is wider. Maybe this is isn't true now, but styling could change.
To make this assertion more future-proof we can do this:

```tsx
const childRect = childEl.getBoundingClientRect();
const parentRect = $markdown[0].getBoundingClientRect();

expect(parentRect.left).to.be.lte(childRect.left);
expect(parentRect.right).to.be.gte(childRect.right);
```